From d0f9057e072d00caac4e7c980b53f093fe9f3aca Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Fri, 17 Aug 2018 14:23:57 -0700
Subject: [PATCH 05/19] vendor/lineage: Add support for java source overlays

*) Set JAVA_SOURCE_OVERLAYS in a device makefile as follows:
   # Java sourcefile overlays (for Android.bp built modules only)
   # Format is a whitespace separated set of rules, each of which
   # structured as follows:
   # modulename|overlay directory|glob pattern within overlay dir
   JAVA_SOURCE_OVERLAYS := \
           org.lineageos.hardware|$(LOCAL_PATH)/lineagehw|**/*.java

*) As of this commit, java sources overlays are only allowed for
   module org.lineageos.hardware.

Change-Id: I6be1c12567081357f5231a84df98ac002c0563b4

# Conflicts:
#	build/soong/soong_config.mk
---
 build/soong/Android.bp          |  1 +
 build/soong/android/config.go   | 12 ++++++++++++
 build/soong/android/variable.go |  1 +
 build/soong/soong_config.mk     |  1 +
 4 files changed, 15 insertions(+)
 create mode 100644 build/soong/android/config.go

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 21304ee0..18e9a841 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -2,6 +2,7 @@ bootstrap_go_package {
     name: "soong-lineage",
     pkgPath: "lineage/soong/android",
     srcs: [
+        "android/config.go",
         "android/variable.go",
     ],
 }
diff --git a/build/soong/android/config.go b/build/soong/android/config.go
new file mode 100644
index 00000000..2ce127d8
--- /dev/null
+++ b/build/soong/android/config.go
@@ -0,0 +1,12 @@
+package android
+
+// Global config used by Lineage soong additions
+var LineageConfig = struct {
+	JavaSourcesOverlayModuleWhitelist []string
+}{
+	[]string{
+		// List of packages that are permitted
+		// for java sources overlay.
+		"org.lineageos.hardware",
+	},
+}
diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index c2bb782a..e3bd1dda 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -26,6 +26,7 @@ type Product_variables struct {
 
 type ProductVariables struct {
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
+	Java_Source_Overlays *string `json:",omitempty"`
 	Needs_text_relocations  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Target_shim_libs  *string `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 577d29d4..7ac293f4 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -4,6 +4,7 @@ _contents := $(_contents)    "Lineage":{$(newline)
 
 # See build/core/soong_config.mk for the add_json_* functions you can use here.
 $(call add_json_bool, Has_legacy_camera_hal1, $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
+$(call add_json_str,  Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
 $(call add_json_bool, Needs_text_relocations, $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_str, Target_shim_libs, $(TARGET_LD_SHIM_LIBS))
-- 
2.17.1

